src/web/demo.rkt

```diff
@@ -453,13 +422,19 @@
       (define formula-str (hash-ref post-data 'formula))
       (define formula (read-syntax 'web (open-input-string formula-str)))
       (define seed* (hash-ref post-data 'seed))
-      (define job-id (compute-job-id (list 'sample formula seed*)))
-      (run-work #t job-id (list 'sample formula seed*))
-      (define result (hash-ref *completed-jobs* job-id))
-      (define pctx (job-result-backend result))
+      (define command (run-herbie-command 'sample formula seed* #f #f #t))
       (define test (parse-test formula))
-      (hasheq 'points (pcontext->json pctx 
-       (context-repr (test-context test)))))))
+      (process-command command (curry post-sample test)))))
+
+(define (post-sample test result)
```

Comment:
I don't understand why this has to be a callback that gets passed around through everything? All that these `post-xxx` commands ever do is pull stuff out of the `job-result-backend`; why not just return that from `run-work` and have this not be a callback?

Reply:
Thanks I will give that a try.